### PR TITLE
Change free space calculation

### LINKED ISSUE #22475: Trash get fully emptied when Cron runs (v18.0.8 and v19.0.2)
I updated yesterday to 18.0.8. My quota is unlimited (it shows 0 B). All my trashbin (deleted files) have been definitely removed.

Step to reproduce:
- Delete a file test.txt
- Go to deleted files in the web UI: the file is there
- Force the cron job to execute (sudo -u www-data php -f /var/www/html/nextcloud/cron.php)
- GO to deleted files: no file anymore

In the log, I see:
"remove "test.txt" (0B) to meet the limit of trash bin size (50% of available quota)"

Am I the only one with this critical issue?

### PR DESCRIPTION
**Pull request by kesselb**
Close #22475 

If neither `$userTrashbinSize` nor `$systemTrashbinSize` are defined we have: `$configuredTrashbinSize = (-1 < 0) ? -1 : -1;`,  `(bool)-1` is true and the freespace is always`-1 - $trashbinSize`.

That should work but no idea how it should be ;)

### CODE REVIEW

**Review by rullzer — APPROVED** [review]
Good catch

**Review by ChristophWurst — APPROVED** [review]

**Review by go2sh — APPROVED** [review]

**Comment by kesselb** [comment]
Are the trashbin tests failing on master too?

File: apps/files_trashbin/tests/TrashbinTest.php [hunk 1]
```diff
@@ -124,15 +124,25 @@ protected function setUp(): void {
...
 			->method('getSystemValue')
-			->willReturnCallback(function ($key, $default) use ($config) {
+			->willReturnCallback(static function ($key, $default) use ($config) {
 				if ($key === 'filesystem_check_changes') {
 					return \OC\Files\Cache\Watcher::CHECK_ONCE;
 				} else {
 					return $config->getSystemValue($key, $default);
 				}
 			});
+		$mockConfig
```
Comment (kesselb) [inline apps/files_trashbin/tests/TrashbinTest.php:136]: Mockageddon :see_no_evil:

**Review by icewind1991 — APPROVED** [review]

**Pull request merged**